lib/member_name_parts.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+require 'scraped'
+
+class MemberNameParts < Scraped::HTML
+  field :prefix do
+    partitioned.first.join(' ')
+  end
+
+  field :name do
+    # .sub called for specific case where Dr. is not separated from name by space
+    partitioned.last.join(' ').sub('Dr.Bharrat', 'Bharrat')
```

Comment:
No — keeping them separate is good. We can hold our noses at this approach then, as long as it's well commented in-place that it's a temporary work around, with a pointer at the issue.